staging/src/k8s.io/kubectl/pkg/util/completion/completion.go

```diff
@@ -94,6 +94,33 @@ func PodResourceNameCompletionFunc(f cmdutil.Factory) func(*cobra.Command, []str
 	}
 }
 
+// PodResourceNameAndContainerPortCompletionFunc Returns a completion function that completes, as a first argument:
+// 1- pod names that match the toComplete prefix
+// 2- resource types containing pods which match the toComplete prefix
+// and as a second argument the containers' ports within the specified pod.
+func PodResourceNameAndContainerPortCompletionFunc(f cmdutil.Factory) func(*cobra.Command, []string, string) ([]string, cobra.ShellCompDirective) {
+	return func(_ *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+		var comps []string
+		var resourceType string
+		directive := cobra.ShellCompDirectiveNoFileComp
+		if len(args) == 0 {
+			comps, directive = doPodResourceCompletion(f, toComplete)
+		} else if len(args) == 1 {
+			t := strings.Split(args[0], "/")
+			// if we specify directly the pod, then resource type is pod, otherwise it would be the resource type passed
+			// by the user
+			if len(t) == 1 {
+				resourceType = "pod"
+			} else {
+				resourceType = t[1]
+			}
+			resourceName := convertResourceNameToPodName(f, args[0])
+			comps = CompGetContainersPorts(f, resourceType, resourceName, toComplete)
```

Comment:
There are various use cases for the port-forward command. I'm curious about this completion covers all type of examples https://github.com/kubernetes/kubernetes/blob/3d56ff21fd3c9c9da82ff22044691ef0671ac7b6/staging/src/k8s.io/kubectl/pkg/cmd/portforward/portforward.go#L74-L95

Reply:
for example, if I run `kubectl port-forward pod/mypod <TAB>` and I want to run `kubectl port-forward service/myservice 8443:https `, how this PR reacts to this?. Because 8443 is the local port number and `https` is the server port number. 

I think, the expected behavior would be;
I should write `8443:` and can execute `<TAB>` and it will complete `https`. Is that correct?

Reply:
Makes sense. Will update that.

Reply:
It should be noted that the current PR works with container port number rather than name. If you see that it's better to work with port name lmk.